lib/cadet/assessments/assessments.ex

```diff
@@ -753,6 +809,69 @@ defmodule Cadet.Assessments do
     |> Repo.transaction()
   end
 
+  def update_contest_voting_submission_status(submission = %Submission{}, question = %Question{}) do
+    count_entries =
+      length(
+        SubmissionVotes
+        |> where(question_id: ^question.id)
+        |> where(user_id: ^submission.student_id)
+        |> Repo.all()
+      )
+
+    count_not_nil_entries =
+      length(
+        SubmissionVotes
+        |> where(question_id: ^question.id)
+        |> where(user_id: ^submission.student_id)
+        |> where([c], not is_nil(c.score))
+        |> Repo.all()
+      )
+
+    if count_entries == count_not_nil_entries do
+      submission |> Submission.changeset(%{status: :attempted}) |> Repo.update()
+    end
+  end
+
+  defp load_contest_voting_entries(questions, assessment_id, user_id) do
+    Enum.map(
+      questions,
+      fn q ->
+        if q.type == :voting do
+          question =
+            Question
+            |> where(assessment_id: ^assessment_id)
+            |> Repo.one()
+
+          voting_question =
+            case all_submission_votes_by_question_id_and_user_id(question.id, user_id) do
+              {:ok, submission_votes} -> Map.put(q.question, :contestEntries, submission_votes)
+            end
```

Comment:
Case with a single match, please just use a direct destructure

(I'm wondering why Credo isn't warning on this)

```elixir
{:ok, submission_votes} = all_submission_votes_by_question_id_and_user_id(question.id, user_id)
voting_question = Map.put(q.question, :contestEntries, submission_votes)
```